### ISSUE TITLE
disable console screen resizing

### ISSUE THREAD

**Issue by goblinfactory** [goblinfactory/konsole#33]
When running a console executable in windows, if you resize the console window while the app is running it totally wrecks the layout. Like below ... (this is not a bug with Konsole, it's a behavior of windows consoles, there's a checkbox setting in options, that is on by default to "wrap text on resize". ) Using platform detection it might be possible to provide a switch to lock the window from being resized. Alternatively, launch a custom console emulator? This issue goes hand in hand with issue #32 as a core requirement for Konsole to be able to be used to build commercial applications and utilities. If the speed of Konsole can be dramatically improved, then detecting resize and offering a Redraw facility similar to Winforms then becomes  a realistic option. 
 
![Capture](https://user-images.githubusercontent.com/1073487/70180178-69fa1c00-16d7-11ea-8251-5239398f5b57.PNG)

![Capture](https://user-images.githubusercontent.com/1073487/70180659-5d29f800-16d8-11ea-8b01-a6dc6c819045.PNG)

**Comment by goblinfactory** [comment]
alternatively, see if we can simply have this value unchecked as part of the console app setting.

**Comment by goblinfactory** [comment]
blows me away that this is not available in .net ? 
https://docs.microsoft.com/en-us/windows/console/setconsolemode

**Comment by goblinfactory** [comment]
Quick update : @eiredrake has confirmed in closed issue #28 that adding the following code does stop a console app from being resized; That looks like a plausible option, just need a nice .net core (.net standard) way to deliver this with the nuget package.

```c
class Program
    {
        private const int MF_BYCOMMAND = 0x00000000;
        public const int SC_CLOSE = 0xF060;
        public const int SC_MINIMIZE = 0xF020;
        public const int SC_MAXIMIZE = 0xF030;
        public const int SC_SIZE = 0xF000;

        [DllImport("user32.dll")]
        public static extern int DeleteMenu(IntPtr hMenu, int nPosition, int wFlags);

        [DllImport("user32.dll")]
        private static extern IntPtr GetSystemMenu(IntPtr hWnd, bool bRevert);

        [DllImport("kernel32.dll", ExactSpelling = true)]
        private static extern IntPtr GetConsoleWindow();

        static void Main(string[] args)
        {
            IntPtr handle = GetConsoleWindow();
            IntPtr sysMenu = GetSystemMenu(handle, false);

            if (handle != IntPtr.Zero)
            {
                DeleteMenu(sysMenu, SC_CLOSE, MF_BYCOMMAND);
                DeleteMenu(sysMenu, SC_MINIMIZE, MF_BYCOMMAND);
                DeleteMenu(sysMenu, SC_MAXIMIZE, MF_BYCOMMAND);
                DeleteMenu(sysMenu, SC_SIZE, MF_BYCOMMAND);
            }
            Console.Read();
        }

    }

```

code is from : https://social.msdn.microsoft.com/Forums/vstudio/en-US/1aa43c6c-71b9-42d4-aa00-60058a85f0eb/c-console-window-disable-resize?forum=csharpgeneral

**Comment by eiredrake** [comment]
Interesting. I'll poke that now.

Yep... that prevents it from going wacky. very cool.

**Comment by goblinfactory** [comment]
@eiredrake  Just released version `4.1` which includes new method `.LockConsoleResizing()` on windows class. Give that a go. Closing this issue.

**Comment by The-Futurist** [comment]
> Quick update : [@eiredrake](https://github.com/eiredrake) has confirmed in closed issue [#28](https://github.com/goblinfactory/konsole/issues/28) that adding the following code does stop a console app from being resized; That looks like a plausible option, just need a nice .net core (.net standard) way to deliver this with the nuget package.
> 
> class Program
>     {
>         private const int MF_BYCOMMAND = 0x00000000;
>         public const int SC_CLOSE = 0xF060;
>         public const int SC_MINIMIZE = 0xF020;
>         public const int SC_MAXIMIZE = 0xF030;
>         public const int SC_SIZE = 0xF000;
> 
>         [DllImport("user32.dll")]
>         public static extern int DeleteMenu(IntPtr hMenu, int nPosition, int wFlags);
> 
>         [DllImport("user32.dll")]
>         private static extern IntPtr GetSystemMenu(IntPtr hWnd, bool bRevert);
> 
>         [DllImport("kernel32.dll", ExactSpelling = true)]
>         private static extern IntPtr GetConsoleWindow();
> 
>         static void Main(string[] args)
>         {
>             IntPtr handle = GetConsoleWindow();
>             IntPtr sysMenu = GetSystemMenu(handle, false);
> 
>             if (handle != IntPtr.Zero)
>             {
>                 DeleteMenu(sysMenu, SC_CLOSE, MF_BYCOMMAND);
>                 DeleteMenu(sysMenu, SC_MINIMIZE, MF_BYCOMMAND);
>                 DeleteMenu(sysMenu, SC_MAXIMIZE, MF_BYCOMMAND);
>                 DeleteMenu(sysMenu, SC_SIZE, MF_BYCOMMAND);
>             }
>             Console.Read();
>         }
> 
>     }
> code is from : https://social.msdn.microsoft.com/Forums/vstudio/en-US/1aa43c6c-71b9-42d4-aa00-60058a85f0eb/c-console-window-disable-resize?forum=csharpgeneral

Sadly this doesn't work when we're running the console under Windows Terminal. I don't think I've ever seen anything as messed up Windows console sessions, its a cluster f**k.

**Comment by goblinfactory** [comment]
Yup ... it's not pretty.
